TMGToolbox/src/input_output/import_binary_matrix.py

```diff
@@ -182,9 +204,17 @@ def __call__(self, xtmf_MatrixType, xtmf_MatrixNumber, ImportFile, xtmf_Scenario
     def _Execute(self):
         with _m.logbook_trace(name="%s v%s" %(self.__class__.__name__, self.version), \
                               attributes= self._GetAtts()):
-            matrix = _util.initializeMatrix(self.MatrixId)
-            if self.MatrixDescription:
-                matrix.description = self.MatrixDescription
+
+            if self.MatrixId == None:
```

Comment:
You should be using `if self.MatrixDescription is None`; because not all objects override `__eq__()`

Reply:
Thank you for the suggestion. I changed the code as suggested and did a quick test. The tool will use the new matrix if an existing matrix has no description (in the case a user selects an existing matrix and then enter a new matrix id wrongly).

Reply:
Using `obj == None` will work in many cases, but [this is not considered to be good Python coding practice](https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_none.html)

Reply:
Thanks a lot for the tips. I replaced `==` by `is`. My concern is that, won't `self.MatrixId is None` more suitable than `self.MatrixDescription is None` since a matrix may not have a description.

Reply:
Yes, that makes sense given my suggestion with `__init__()`. Using the description as the "sentinel" (signal for creating a matrix) is a poor choice as there is a use case for creating a new matrix without a description.

Reply:
Thank you for confirming. I'll submit the commit accordingly.